kura/test/org.eclipse.kura.core.configuration.test/src/test/java/org/eclipse/kura/core/configuration/ConfigurationServiceTest.java

```diff
@@ -3003,4 +3095,165 @@ String getSnapshotsDirectory() {
 
         assertEquals(expect, new String(chars));
     }
+
+    private Component createMockComponent(final String pid, final String... implementedServices) {
+        final Component result = mock(Component.class);
+        when(result.getName()).thenReturn(pid);
+        when(result.getServices()).thenReturn(implementedServices);
+        return result;
+    }
+
+    private Map<String, Tocd> getOcdsMap(List<String> registeredFactories, List<Tocd> registeredOcds) {
+        final Map<String, Tocd> result = new HashMap<>();
+        for (int i = 0; i < registeredFactories.size(); i++) {
+            result.put(registeredFactories.get(i), registeredOcds.get(i));
+        }
+        return result;
+    }
+
+    private ConfigurationService createMockConfigurationServiceForOCDTests(List<String> registeredFactories,
+            List<Tocd> registeredOcds, List<Component> registeredComponents) throws NoSuchFieldException {
+
+        assertEquals(registeredFactories.size(), registeredOcds.size());
+        ScrService scrService = new ScrService() {
```

Comment:
Since most methods return null, it could be a mock that just reacts to getComponents().

Reply:
Yes definitely, I've tried using a mock here but I was getting some strange "unfinished stubbing" errors at runtime. I did not spend much time investigating and I've used this quick and dirty solution. I will try again and see if I can manage to use a proper mock.